src/validate.js

```diff
@@ -0,0 +1,96 @@
+import createSchema from "graphql-to-json-schema";
+import produce from "immer";
+import * as yup from "yup";
+
+const curry = (f, arr = []) => (...args) =>
+  (a => (a.length >= f.length ? f(...a) : curry(f, a)))([...arr, ...args]);
+
+const migrate = schema =>
+  produce(schema, draft => {
+    draft["$schema"] = "http://json-schema.org/draft-07/schema#";
+  });
+
+const getPropRules = ({ definitions }, rootType) =>
+  Object.entries(definitions[rootType].properties);
+
+const handleType = (config, type, required, ref, items) => {
```

Comment:
If these were named parameters, you could skip the renaming acrobatics in the array handler below and just pass the params through (with the config key, of course).

Reply:
Keep the spec name for `$ref` so we don't need to wonder why we're using two different names for `$ref`.